File: src/com/l2kt/gameserver/model/zone/type/NoStoreZone.java
package com.l2kt.gameserver.model.zone.type;

import com.l2kt.gameserver.model.zone.ZoneId;
import com.l2kt.gameserver.model.zone.ZoneType;
import com.l2kt.gameserver.model.actor.Creature;
import com.l2kt.gameserver.model.actor.instance.Player;

/**
 * A zone extending {@link ZoneType} where store isn't allowed.
 */
public class NoStoreZone extends ZoneType
{
	public NoStoreZone(final int id)
	{
		super(id);
	}
	
	@Override
	protected void onEnter(final Creature character)
	{
		if (character instanceof Player)
			character.setInsideZone(ZoneId.NO_STORE, true);
	}
	
	@Override
	protected void onExit(final Creature character)
	{
		if (character instanceof Player)
			character.setInsideZone(ZoneId.NO_STORE, false);
	}
}

File: src/com/l2kt/gameserver/model/zone/type/NoStoreZone.kt
package com.l2kt.gameserver.model.zone.type

import com.l2kt.gameserver.model.actor.Creature
import com.l2kt.gameserver.model.actor.instance.Player
import com.l2kt.gameserver.model.zone.ZoneId
import com.l2kt.gameserver.model.zone.ZoneType

/**
 * A zone extending [ZoneType] where store isn't allowed.
 */
class NoStoreZone(id: Int) : ZoneType(id) {

    override fun onEnter(character: Creature) {
        if (character is Player)
            character.setInsideZone(ZoneId.NO_STORE, true)
    }

    override fun onExit(character: Creature) {
        if (character is Player)
            character.setInsideZone(ZoneId.NO_STORE, false)
    }
}